starport_template/lib/stores/wallets_store.dart

```diff
@@ -50,4 +65,68 @@ class WalletsStore {
     wallets.value.add(creds.publicInfo);
     return creds.publicInfo;
   }
+
+  Future getCosmosBalances(BaseEnv baseEnv, String walletAddress) async {
+    isError.value = false;
+    isBalancesLoading.value = true;
+    try {
+      final uri = '${baseEnv.baseApiUrl}/cosmos/bank/v1beta1/balances/$walletAddress';
+      final response = await http.get(Uri.parse(uri));
+      final map = response.body as Map<String, dynamic>;
+      final list = map['balances'] as List<Map<String, dynamic>>;
+      var balanceList = <BalanceJson>[];
+      if (list.isNotEmpty) {
+        balanceList = list.map((e) => BalanceJson.fromJson(e)).toList();
+      }
+
+      balancesList.value = balanceList
+          .map(
+            (e) => Balance(
+              denom: Denom(e.denom),
+              amount: Amount.fromString(e.amount),
+            ),
+          )
+          .toList();
```

Comment:
you can also `map` on an empty list, there is no need of having this `if` statement here:
```suggestion
        balancesList.value = list
           .map((e) => BalanceJson.fromJson(e))
          .map(
            (e) => Balance(
              denom: Denom(e.denom),
              amount: Amount.fromString(e.amount),
            ),
          )
          .toList();
```

Reply:
✅